buildrunner/docker/multiplatform_image_builder.py

```diff
@@ -0,0 +1,379 @@
+            for name, images in self._intermediate_built_images.items():
+                for image in images:
+                    for tag in image.tags:
+                        LOGGER.debug(f"Removing image {image.repo}:{tag} for {name}")
+                        docker.image.remove(f'{image.repo}:{tag}', force=True)
+
+        # Removes all tagged images if keep_images is False
+        if self._tagged_images_names and not self._keep_images:
+            for name, images in self._tagged_images_names.items():
+                for image in images:
+                    LOGGER.debug(f"Removing tagged image {image} for {name}")
+                    docker.image.remove(image, force=True)
+
+    @property
+    def registry_ip(self) -> int:
+        """Returns the ip address of the local registry"""
+        return self._reg_ip
+
+    @property
+    def registry_port(self) -> int:
+        """Returns the port of the local registry"""
+        return self._reg_port
+
+    @property
+    def tagged_images_names(self) -> List[str]:
+        """Returns a list of all the tagged images names"""
+        return self._tagged_images_names
+
+    def is_multiplatform(self, name: str) -> bool:
+        """Returns True if the image with name is multiplatform"""
+        return len(self._intermediate_built_images.get(name, [])) > 0
+
+    def registry_address(self) -> str:
+        """Returns the address of the local registry"""
+        return f"{self._reg_ip}:{self._reg_port}"
+
+    def _start_local_registry(self):
+        """
+        Starts a local docker registry container
+
+        Returns:
+            str: The name of the registry container
+        """
+        LOGGER.debug("Starting local docker registry")
+        container = docker.run("registry", detach=True, publish_all=True)
+        ports = container.network_settings.ports
+
+        # Something changed in the registry image, so we need to update this code
+        assert len(ports) == 1, \
+            f"Expected 1 port, but got {len(ports)}"
+        assert isinstance(ports.get('5000/tcp')[0], dict), \
+            f"Expected dict, but got {type(ports.get('5000/tcp')[0])}"
+        assert ports.get('5000/tcp')[0].get('HostIp') == '0.0.0.0', \
+            f"Expected HostIp to be 0.0.0.0 but got {ports.get('5000/tcp')[0].get('HostIp')}"
+
+        self._reg_ip = 'localhost'
+        self._reg_port = ports.get('5000/tcp')[0].get('HostPort')
+        self._reg_name = container.name
+        LOGGER.debug(f"Started local registry {self._reg_name} at {self._reg_ip}:{self._reg_port}")
+
+    def _stop_local_registry(self):
+        """
+        Stops and removes the local registry along with any images
+        """
+        LOGGER.debug(f"Stopping and removing local registry {self._reg_name} at {self._reg_ip}:{self._reg_port}")
+        docker.stop(self._reg_name)
+        docker.remove(self._reg_name, volumes=True, force=True)
+
+    # pylint: disable=too-many-arguments
+    def build_image(self,
```

Comment:
This wasn't clear until I read the code that this method is for building a single platform and the next method is for building all of them. Maybe we can make this a little clearer by the method names?

Reply:
Renamed methods